cmd/poseidon/main.go

```diff
@@ -156,17 +158,36 @@ func runServer(server *http.Server, cancel context.CancelFunc) {
 	defer cancel()
 	defer shutdownSentry() // shutdownSentry must be executed in the main goroutine.
 
-	log.WithField("address", server.Addr).Info("Starting server")
+	// Creating http Listener
+	var httpListener net.Listener
 	var err error
+	if config.Config.Server.SystemdSocketActivation {
+		var listeners []net.Listener
+		listeners, err = activation.Listeners()
+		if len(listeners) > 0 {
+			httpListener = listeners[0]
+		}
+	} else {
+		httpListener, err = net.Listen("tcp", server.Addr)
+	}
+	if err != nil || httpListener == nil {
+		log.WithError(err).
+			WithField("listener", httpListener).
+			WithField("systemd_socket", config.Config.Server.SystemdSocketActivation).
+			Fatal("Failed listening to socket")
+		return
+	}
+
+	// Serving for that http listener
+	log.WithField("address", server.Addr).Info("Starting server")
```

Comment:
The `address` field printed in the log statement is the one given in the configuration. However, if using a Systemd socket, the configuration is not applied. Hence, this is misleading (and should rather reflect the actual `listener`), I think.